hooks/command

```diff
@@ -0,0 +1,118 @@
+# exposes a string value at BUILDKITE_PLUGIN_{NAME}_{KEY}, and array values at
+# BUILDKITE_PLUGIN_{NAME}_{KEY}_{IDX}.
+function read_list_property {
+  local prefix="BUILDKITE_PLUGIN_DYNAMODB_IMAGE_${1}"
+  local property="${prefix}_0"
+  result=()
+  if [[ -n ${!property:-} ]]; then
+    local i=0
+    local property="${prefix}_${i}"
+    while [[ -n ${!property:-} ]]; do
+      result+=("${!property}")
+      i=$((i + 1))
+      property="${prefix}_${i}"
+    done
+  elif [[ -n ${!prefix:-} ]]; then
+    result+=("${!prefix}")
+  fi
+  [[ ${#result[@]} -gt 0 ]] || return 1
+}
+
+# Read tables into an array
+function read_tables {
+  if read_list_property 'TABLES'; then
+    for table in "${result[@]}"; do
+      tables+=("${table}")
+    done
+  else
+    echo "A list of tables must be provided."
+  fi
+}
+
+# Retrieves schemas and saves them as JSON
+function retrieve_schemas {
+  for table in "${tables[@]}"; do
+    local schema_file="${tmp_dir}/${table}.json"
+    aws dynamodb describe-table \
+      --table-name "${table}" \
+      --output json \
+      >"${schema_file}"
+  done
+}
+
+# Pulls local DynamoDB and runs it on port 8000
+function start_local_dynamo {
+  docker pull amazon/dynamodb-local:latest
+  local_dynamo_container_id=$(docker run -d -p $local_dynamo_port:$local_dynamo_port amazon/dynamodb-local:latest -jar DynamoDBLocal.jar -port $local_dynamo_port -sharedDb)
+  sleep 5 # Give container a bit of time to start up
+}
+
+# Saves the database file so that we can use it in the build
+function save_database {
+  docker cp $local_dynamo_container_id:/home/dynamodblocal/shared-local-instance.db $tmp_dir/shared-local-instance.db
+}
+
+# Stop running local dynamo
+function stop_local_dynamo {
+  docker stop $local_dynamo_container_id
+}
+
+# Create the tables
+function create_tables {
+  local local_dynamo_endpoint="http://localhost:$local_dynamo_port"
+  for table in "${tables[@]}"; do
+    local schema_file="${tmp_dir}/${table}.json"
+    local attributes=$(cat "${schema_file}" | jq '.["Table"]."AttributeDefinitions"')
+    local key_schema=$(cat "${schema_file}" | jq '.["Table"]."KeySchema"')
+    local gsis=$(cat "${schema_file}" | jq '[.["Table"]."GlobalSecondaryIndexes"[] | {Projection: .Projection, IndexName: .IndexName, KeySchema: .KeySchema }]')
+    aws dynamodb create-table \
+      --table-name "${table}" \
+      --attribute-definitions "${attributes}" \
+      --key-schema "${key_schema}" \
+      --global-secondary-indexes "${gsis}" \
+      --billing-mode 'PAY_PER_REQUEST' \
+      --region "local" --endpoint ${local_dynamo_endpoint}
+  done
+  aws dynamodb list-tables --region "local" --endpoint ${local_dynamo_endpoint}
+}
+
+# Builds the multi-arch image and publishes it
+function build_and_publish {
+  cd $script_dir
+  docker buildx create --use
+  if [[ ${BUILDKITE_BRANCH:-branch} == "${BUILDKITE_PIPELINE_DEFAULT_BRANCH:-master}" ]]; then
```

Comment:
Same here; do we need the fallbacks or can we eagerly fail?

Reply:
I think we should fail 👍